File: e2e/playwright-typescript/interactions/steps/LanguageSwitcherStep.ts
import { expect, Page } from '@playwright/test';
import { step } from '@/decorators/step-decorator';
import { LanguageSwitcherDropdown } from '@/interactions/pages/LanguageSwitcherDropdown';

export class LanguageSwitcherStep {
  private readonly dropdown: LanguageSwitcherDropdown;

  constructor(page: Page) {
    this.dropdown = new LanguageSwitcherDropdown(page);
  }

  @step("Select language '{language}'")
  async selectLanguage(language: string): Promise<void> {
    await this.dropdown.dropdown.click();
    await this.dropdown.items.filter({ hasText: language }).click();
    await expect(this.dropdown.dropdownValue).toHaveText(language);
  }
}

File: e2e/playwright-python/interactions/steps/language_switcher_step.py
from playwright.sync_api import Page, expect
from interactions.pages.language_switcher_dropdown import LanguageSwitcherDropdown
from decorators.step_decorator import step


class LanguageSwitcherStep:
    def __init__(self, page: Page):
        self._dropdown = LanguageSwitcherDropdown(page)

    @step("Select language '{language}'")
    def select_language(self, language: str) -> None:
        self._dropdown.dropdown.click()
        self._dropdown.items.filter(has_text=language).click()
        expect(self._dropdown.dropdown_value).to_have_text(language)
